bundle/phases/initialize.go

```diff
@@ -26,36 +26,70 @@ func Initialize(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 		// If it is an ancestor, this updates all paths to be relative to the sync root path.
+		// Reads (typed): b.Config.Sync.Paths (calculates longest common parent together with bundle root).
+		// Updates (typed) b.{SyncRoot,SyncRootPath}  (set to calculate sync root, which is either bundle root or some parent of bundle root)
+		// Updates (typed) b.Config.{Sync,Include,Exclude} they set to be relative to SyncRootPath instead of bundle root
 		mutator.SyncInferRoot(),
 
+		// Reads (typed): b.Config.Workspace.CurrentUser (checks if it's already set)
+		// Updates (typed): b.Config.Workspace.CurrentUser (sets user information from API)
+		// Updates (typed): b.Tagging (configures tagging object based on workspace client)
 		mutator.PopulateCurrentUser(),
+
+		// Updates (typed): b.WorktreeRoot (sets to SyncRoot if no git repo found, otherwise to git worktree root)
+		// Updates (typed): b.Config.Bundle.Git.{ActualBranch,Branch,Commit,OriginURL,BundleRootPath} (loads git repository details)
+		// Loads git repository information and updates bundle configuration with git details
 		mutator.LoadGitDetails(),
 
 		// This mutator needs to be run before variable interpolation and defining default workspace paths
 		// because it affects how workspace variables are resolved.
 		mutator.ApplySourceLinkedDeploymentPreset(),
 
+		// Reads (typed): b.Config.Workspace.RootPath (checks if it's already set)
+		// Reads (typed): b.Config.Bundle.Name, b.Config.Bundle.Target (used to construct default path)
+		// Updates (typed): b.Config.Workspace.RootPath (sets to ~/.bundle/{name}/{target} if not set)
 		mutator.DefineDefaultWorkspaceRoot(),
+
+		// Reads (typed): b.Config.Workspace.RootPath (checks if it's already set)
+		// Reads (typed): b.Config.Workspace.CurrentUser (used to expand ~ in path)
+		// Updates (typed): b.Config.Workspace.RootPath (expands ~ to user's home directory if present)
 		mutator.ExpandWorkspaceRoot(),
+
+		// Reads (typed): b.Config.Workspace.RootPath (used to construct default paths)
+		// Updates (typed): b.Config.Workspace.{FilePath,ResourcePath,ArtifactPath,StatePath} (sets default paths if not already set)
 		mutator.DefineDefaultWorkspacePaths(),
+
+		// Reads (dynamic): workspace.{root_path,file_path,artifact_path,state_path,resource_path} (reads paths to prepend prefix)
+		// Updates (dynamic): workspace.{root_path,file_path,artifact_path,state_path,resource_path} (prepends "/Workspace" to paths that don't already have specific prefixes)
 		mutator.PrependWorkspacePrefix(),
 
-		// This mutator needs to be run before variable interpolation because it
-		// searches for strings with variable references in them.
+		// Reads (dynamic): * (strings) (searches for strings with workspace path variables prefixed with "/Workspace")
+		// Updates (dynamic): * (strings) (removes "/Workspace" prefix from workspace path variables)
+		// Finds and removes "/Workspace" prefix from workspace path variables in string values
```

Comment:
This confused me because It's not actually correct, it does not removes prefix from variables

Reply:
thanks, corrected